### PR TITLE
docs: reorganize documentation directory structure

### PR DESCRIPTION
**Pull request by ovidiutirla**
**What this PR does / why we need it**: 
Fixes https://github.com/kubernetes/kube-state-metrics/issues/1903

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
no impact

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://github.com/kubernetes/kube-state-metrics/issues/1903

## Testing

```
$ make doccheck 
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags "-s -w -X github.com/prometheus/common/version.Version=v2.11.0 -X github.com/prometheus/common/version.Revision=fc204cb8 -X github.com/prometheus/common/version.Branch=reorg-docs -X github.com/prometheus/common/version.BuildUser=otirla@otirla-ct.c.googlers.com -X github.com/prometheus/common/version.BuildDate=2024-04-03T10:30:58Z" -o kube-state-metrics
gomplate -d config=./data.yaml --file README.md.tpl > README.md
>> generating docs
embedmd -w `find . -path ./vendor -prune -o -name "*.md" -print`
git diff --no-ext-diff --quiet --exit-code README.md
- Checking if the generated documentation is up to date...
- Checking if the documentation is in sync with the code...
OK
- Checking for orphan documentation files
OK
```

### CODE REVIEW

File: docs/metrics/extend/horizontalpodautoscaler-metrics.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
...
-| kube_horizontalpodautoscaler_labels                  | Gauge       | Kubernetes labels converted to Prometheus labels controlled via [--metric-labels-allowlist](./cli-arguments.md)           | `horizontalpodautoscaler`=&lt;hpa-name&gt; <br> `namespace`=&lt;hpa-namespace&gt;                                                                                                                                                                        | STABLE       |
+| kube_horizontalpodautoscaler_annotations             | Gauge       | Kubernetes annotations converted to Prometheus labels controlled via [--metric-annotations-allowlist](../../developer/cli-arguments.md) | `horizontalpodautoscaler`=&lt;hpa-name&gt; <br> `namespace`=&lt;hpa-namespace&gt;                                                                                                                                                                        | EXPERIMENTAL |
```
Comment (CatherineF-dev) [inline docs/metrics/extend/horizontalpodautoscaler-metrics.md:6]: should HPA be moved into workload?

**Comment by CatherineF-dev** [comment]
overall lgtm. Added few comments.

File: docs/metrics/extend/horizontalpodautoscaler-metrics.md (hunk 1 above)
Comment (ovidiutirla) [reply docs/metrics/extend/horizontalpodautoscaler-metrics.md:6]: Yes, moving it now to workload.

**Comment by ovidiutirla** [comment]
hey @rexagod, I'm happy to address any other feedback. Thank you.

**Pull request merged**